pages/api/video.ts

```diff
@@ -0,0 +1,118 @@
+  });
+
+  await findPaints;
+  console.log("found points", gPaints.length);
+
+  return gPaints;
+}
+
+async function fetchGraphics(
+  tmpDir: string,
+  paintPoints: PaintPoint[],
+  { client, sessionId }: { client: ProtocolClient; sessionId: string }
+) {
+  let prevTime = 0;
+  let input = "";
+  const numPaints = paintPoints.length;
+  await Promise.all(
+    paintPoints.map(async (point, index) => {
+      const duration = (point.time - prevTime) / 1000;
+      const filename = path.join(tmpDir, `/${index}-${point.point}-${point.time}.jpeg`);
+      input += `file '${filename}'\nduration ${duration}\n`;
+
+      prevTime = point.time;
+
+      const { screen } = await client.Graphics.getPaintContents(
+        { point: point.point, mimeType: point.screenShots[0].mimeType },
+        sessionId
+      );
+      console.log(`${index}/${numPaints} - ${filename}`);
+      fs.writeFileSync(filename, Buffer.from(screen.data, "base64"));
+    })
+  );
+
+  const inputFilename = path.join(tmpDir, "/input.txt");
+  console.log(inputFilename);
+  fs.writeFileSync(inputFilename, input);
+  return inputFilename;
+}
+
+async function createVideo(tmpDir: string, input: string, recordingId: string): Promise<string> {
+  const { promise, resolve, reject } = defer<string>();
+  const filename = path.join(tmpDir, `/${recordingId}.mp4`);
+
+  console.log(`starting ffmpeg`);
+  const proc = ffmpeg();
+  proc.input(input).inputOptions(["-safe 0"]).inputFormat("concat");
+
+  proc
+    .on("end", () => {
+      console.log("Finished processing");
+      resolve(filename);
+    })
+    .on("error", (err: any) => {
+      console.log("Error:", err);
+      reject(err);
+    })
+    .outputOptions(["-c:v libx264", "-r 30", "-pix_fmt yuv420p"])
+    .save(filename);
+
+  return promise;
+}
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { recordingId, token } = req.body;
+  console.log(`creating video for ${recordingId}...`);
+
+  try {
+    const { client } = await createClient();
+    await client.Authentication.setAccessToken({ accessToken: token });
+
+    const { sessionId } = await client.Recording.createSession({ recordingId });
+    console.log(`sessionId: ${sessionId}`);
+
+    const tmpDir = os.tmpdir();
+    const paintPoints = await fetchPaintPoints({ client, sessionId });
+    const inputFilename = await fetchGraphics(tmpDir, paintPoints, { client, sessionId });
+
+    const video = await createVideo(tmpDir, inputFilename, recordingId);
+    console.log(sessionId, inputFilename, video);
+    //   res.status(200).json({ tmpDir, recordingId, sessionId, inputFilename, video });
+
+    const fileStream = fs.createReadStream(video);
+    const filename = path.basename(video);
+
+    res.setHeader("Content-Type", "video/mp4");
+    res.setHeader("Content-Disposition", `inline; filename="${filename}"`);
+
+    fileStream.pipe(res);
+  } catch (error) {
+    console.log(error);
+    res.status(500).send(error);
```

Comment:
We might not want to send the entire error to the client because it could leak internal information. Just logging the error should be OK.